shuffle/cli/src/deploy.rs

```diff
@@ -53,65 +43,223 @@ fn publish_packages_as_transaction(
+    let mut new_account = LocalAccount::new(derived_address, new_account_key, account_seq_number);
+
+    let mut all_module_names = Vec::new();
+    let mut all_hashes = Vec::new();
+
+    let compiled_package = build_move_package(project_path.join(MAIN_PKG_PATH).as_ref())?;
 
-pub fn send_module_transaction(
-    compiled_package: &CompiledPackage,
-    client: &BlockingClient,
-    account: &mut LocalAccount,
-    factory: &TransactionFactory,
-) -> Result<()> {
     for module in compiled_package
         .transitive_compiled_modules()
         .compute_dependency_graph()
         .compute_topological_order()?
     {
         let module_id = module.self_id();
-        if module_id.address() == &account.address() {
-            println!("Deploying Module: {}", module_id);
-            let mut binary = vec![];
-            module.serialize(&mut binary)?;
-            let publish_txn = account.sign_with_transaction_builder(
-                factory.payload(TransactionPayload::Module(Module::new(binary))),
+        if module_id.address() != &new_account.address() {
+            println!("Skipping Module: {}", module_id);
+            continue;
+        }
+        println!("Deploying Module: {}", module_id);
+        let mut binary = vec![];
+        module.serialize(&mut binary)?;
+
+        let hash = send_module_transaction(&client, &mut new_account, binary).await?;
+
+        all_module_names.push(module_id.name().as_str().to_string());
+        all_hashes.push(hash);
+    }
+
+    confirm_txns_executed_from_hashes(&client, all_hashes).await?;
+    confirm_modules_exist(&client, all_module_names, derived_address).await
+}
+
+async fn send_module_transaction(
+    client: &DevApiClient,
+    account: &mut LocalAccount,
+    module_binary: Vec<u8>,
+) -> Result<String> {
+    let factory = TransactionFactory::new(ChainId::test());
+    let publish_txn = account.sign_with_transaction_builder(
+        factory.payload(TransactionPayload::Module(Module::new(module_binary))),
+    );
+    let bytes = bcs::to_bytes(&publish_txn)?;
+    let resp = client.post_transactions(bytes).await?;
+    let json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+    let hash = get_hash_from_post_txn(json)?;
+    Ok(hash)
+}
+
+pub async fn confirm_txns_executed_from_hashes(
+    client: &DevApiClient,
+    all_hashes: Vec<String>,
+) -> Result<()> {
+    for hash in all_hashes.iter() {
+        let mut resp = client.get_transactions_by_hash(hash.as_str()).await?;
+        let mut json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+        let start = Instant::now();
+        while json["type"] == "pending_transaction" {
+            resp = client.get_transactions_by_hash(hash.as_str()).await?;
+            json = serde_json::from_str(resp.text().await?.as_str())?;
+            let duration = start.elapsed();
+            if duration > Duration::from_secs(10) {
```

Comment:
What's going on here? Are you spinning in the while loop until 10 seconds pass or json type is != `pending_transaction`? That's a lot of CPU spins, let's sleep a bit.

Reply:
I'm spinning in the while loop until  json type is != `pending_transaction`. If for whatever reason, json type stays at pending_transactions, after 10 seconds we break out of the loop. Will add a sleep here too.